cogs/utils/image_helpers.py

```diff
@@ -0,0 +1,194 @@
+import discord
+from discord.ext import commands
+
+# misc imports
+import numpy as np
+import cv2
+import concurrent.futures
+import math
+from functools import partial
+from io import BytesIO
+
+
+def apply_transform(transform, buffer, size, max_size, ext, is_png, *args):
+    img_bytes = np.asarray(bytearray(buffer.read()), np.uint8)
+    result = cv2.imdecode(img_bytes, cv2.IMREAD_UNCHANGED)
+    ratio = (max_size / size) * 100 if size > max_size else None
+
+    if ratio:
+        _, buffer = cv2.imencode(f".{ext}", result,
+                                 (cv2.CV_IMWRITE_PNG_COMPRESSION if is_png else
+                                  cv2.IMWRITE_JPEG_QUALITY, ratio))
+        result = cv2.imdecode(buffer, cv2.IMREAD_UNCHANGED)
+
+    result = transform(result, *args)
+
+    _, buffer = cv2.imencode(f".{ext}", result)
+
+    return buffer
+
+
+async def fitler_image(loop, transform, ctx, history_limit, max_size, *args):
+
+    att = await get_attachment(ctx, history_limit)
+    if att is None:
+        await ctx.send("no image could be found (only attached image files"
+                       " can be detected) or message could not be found")
+        return
+
+    await ctx.trigger_typing()
+    buffer = BytesIO()
+    await att.save(fp=buffer)
+
+    original_name, ext = att.filename.rsplit(".", 1)
+    ext = ext.lower()
+    if ext in ("jpeg", "jpg"):
+        is_png = False
+    elif ext in ("png"):
```

Comment:
this is not correct, since `("png")` is not a tuple - should be `== "png"` or `in ("png",)` but i think the former is better